src/router/helpersClass.ts

```diff
@@ -37,22 +37,27 @@ export function getEffectivePriceSwapForPath(
     path: NewPath,
     swapType: SwapTypes,
     amount: OldBigNumber,
-    inputDecimals: number
+    inputDecimals: number,
+    costReturnToken: BigNumber
 ): OldBigNumber {
     if (amount.lt(INFINITESIMAL)) {
         // Return spot price as code below would be 0/0 = undefined
         // or small_amount/0 or 0/small_amount which would cause bugs
         return getSpotPriceAfterSwapForPath(path, swapType, amount);
     }
-    const outputAmountSwap = getOutputAmountSwapForPath(
+
+    let outputAmountSwap = getOutputAmountSwapForPath(
         path,
         swapType,
         amount,
         inputDecimals
     );
+    const gasCost = bnum(costReturnToken.toString()).times(path.pools.length);
```

Comment:
@sergioyuhjtman - just wanted to double check we're using the correct scale here? 
I see [here](https://github.com/balancer-labs/balancer-sor/blob/5a24d07cd8551ca394df439176a9b2a8dc126f04/src/router/sorClass.ts#L101) it scales to floating point and wondered if we need to do similar?